File: robot.cpp
class Solution {
public:
    bool robot(string command, vector<vector<int>>& obstacles, int x, int y) {
        if(obstacles.size()==0||(x==0&&y==0)) return true;
        int cor_x = 0;
        int cor_y = 0;
        int i = 0;
        map<int,vector<int>> dic;
        for(int j=0;j<obstacles.size();j++){
            int index1 = obstacles[j][0];
            int index2 = obstacles[j][1];
            dic[index1].push_back(index2);
        }
        while(x!=cor_x||y!=cor_y){
            if(command[i%command.size()]=='U'){
                cor_y += 1;
            }
            else if(command[i%command.size()]=='R'){
                cor_x += 1;
            }
            if(cor_x>x||cor_y>y){
                return false;
            }
            
            if(dic.find(cor_x)!=dic.end()){
                if(find(dic[cor_x].begin(),dic[cor_x].end(),cor_y)!=dic[cor_x].end()){
                    return false;
                }
            }
            i++;
        }
        return true;
    }
};

File: robot.py
class Solution:
    def robot(self, command: str, obstacles: List[List[int]], x: int, y: int) -> bool:
        if len(obstacles) == 0 or (x == 0 and y == 0):
             return True
        if command == '':
            return False
        cor_x = 0
        cor_y = 0
        i = 0
        dic = {}
        for point in obstacles:
            if point[0] not in dic.keys():
                dic[point[0]] = [point[1]]
            else:
                dic[point[0]].append(point[1])
        while cor_x!=x or cor_y!=y:
            if command[i%len(command)] == 'U':
                cor_y += 1
            elif command[i%len(command)] == 'R':
                cor_x += 1
            if cor_x>x or cor_y>y:
                return False
            if cor_x in dic.keys() and cor_y in dic[cor_x]: 
                return False
            i+=1
        return True
